### ISSUE TITLE
Code Table Request - migrate larva from age class to life stage

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#7521]
### Goal
_Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step._

Move terms deemed to be life stages to the newly created code table. This process will be completed by the DBA.
1. all current larva age class attributes will be converted to larva life stage attributes
2. larva will no longer be available in age class
3. all collections with larva in use at the time this process is completed will have the attribute added as "usedby" in the code table

### Context
_Describe why this new value is necessary and existing values are not._

#7298 

### Table
_Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure._

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctlife_stage

### Proposed Value
_Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos._

larva

### Proposed Definition
_Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable._

Current definition in age class:

~~The active immature form of an insect, especially one that differs greatly from the adult and forms the stage between egg and pupa, e.g. a caterpillar or grub. OR An immature form of other animals that undergo some metamorphosis, e.g. a tadpole.~~

Replace with new definition in life stage:

A larva (Latin; plural larvae) is a young (juvenile) form of animal with indirect development, going through or undergoing metamorphosis (for example, insects, amphibians, or cnidarians). The larva can look completely different from the adult form, for example, a caterpillar differs from a butterfly. Larvae often have special (larval) organs which do not occur in the adult form. The larvae of some species can become pubescent and not further develop into the adult form (for example, in some newts). This is a type of neoteny. It is a misunderstanding that the larval form always reflects the group's evolutionary history. It could be the case, but often the larval stage has evolved secondarily, as in insects. In these cases the larval form might differ more from the group's common origin than the adult form. The early life stages of most fish species are considerably different from juveniles and adults of their species and are called larvae. | http://rs.gbif.org/vocabulary/gbif/life_stage/larva

### Best For
_Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm_

[Amph](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#amph)
[Arc](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#arc)
[Bird](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#bird)
[Egg](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#egg)
[Ento](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#ento)
[Env](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#env)
[ES](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#es)
[Fish](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#fish)
[Herp](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#herp)
[Hild](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#hild)
[Host](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#host)
[Inv](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#inv)
[Mala](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#mala)
[Para](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#para)
[Rept](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#rept)
[Teach](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#teach)
[Vert](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#vert)
[Zoo](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcollection_cde#zoo)

### Search Terms

subadult,juvenile,larvae,tadpole,polliwog,pollywog,polliwig,polewig,polwig,planula,nauplius,zoea,nymph,caterpillar,grub,maggot,wriggler,trochophore,veliger,glochidium,ammocoete,leptocephalus,bipinnaria,caterpillar,grub,maggot

#### Attribute data type
_If the request is for an attribute, what values will be allowed?
free-text, categorical, or number+units depending upon the attribute (TBA)_

#### Attribute controlled values
_If the values are categorical (to be controlled by a code table), add a link to the appropriate code table. If a new table or set of values is needed, please elaborate._

#### Attribute units
_if numerical values should be accompanied by units, provide a link to the appropriate units table._

#### Part preservation attribute affect on "tissueness"
_if a new part preservation is requested, please add the affect it would have on "tissueness": No Influence, Allows, or Denies_

### Priority
_Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking._

### Example Data
_Requests with clarifying sample data are generally much easier to understand and prioritize. Please attach or link to any representative data, in any form or format, which might help clarify the request._

### Available for Public View
_Most data are by default publicly available. Describe any necessary access restrictions._

### Helpful Actions

- [ ] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [ ] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [ ] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [ ] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
4. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
5. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [ ] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by dustymc** [comment]
I don't particularly like the definition; it's redundant with search_terms, and I don't think very functional (eg it won't help me decide if a specimen needs the label). Here's a simpler proposal that I think won't find creative ways to get complicated in the future (eg when perhaps our understanding of "cnidarians" is wildly different than now):

>The immature form of an animal.

The [current](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctage_class#larva) definition is 

>The active immature form of an insect, especially one that differs greatly from the adult and forms the stage between egg and pupa, e.g. a caterpillar or grub. OR An immature form of other animals that undergo some metamorphosis, e.g. a tadpole.

I don't think anything proposed fundamentally changes that; this is and will remain a move to a more inclusive container with no change to the data.

-------------

Here's data:
[temp_larva.csv.zip](https://github.com/ArctosDB/arctos/files/14564734/temp_larva.csv.zip)

Summary:

```

 guid_prefix  | count 
--------------+-------
 UCSC:Herp    |     2
 CHAS:Herp    |    89
 UWBM:Herp    |   348
 UTEP:Herp    |   268
 UCM:Obs      |     2
 UAMObs:Ento  |   309
 OGL:Genomic  |   344
 KNWR:Ento    |    39
 ALMNH:Ento   |     1
 UMZM:Para    |     1
 MVZ:Herp     | 11848
 UNR:Fish     |     1
 UNR:Herp     |    70
 NMMNH:Paleo  |     1
 UNCG:Para    |    57
 NMU:Para     |     7
 DMNS:Inv     |     2
 PSM:Paleo    |    57
 USNPC:Para   |    63
 ASUMZ:Invert |   559
 OWU:Inv      |     1
 UMNH:Herp    |     7
 UAM:Ento     |  7903
 MSB:Herp     |   966
 UCM:Herp     |   244
 UTEP:HerpOS  |    20
 APSU:Fish    |    22
 UCM:Fish     |     1
 OWU:Amph     |     1
 CHAS:Inv     |     2
 UWZM:Herp    |   222
 CSULB:Herp   |     7
 HWML:Para    |  5501
 UAM:Herp     |     3
 UWYMV:Herp   |    19
 MSB:Para     |  3717
 MLZ:Fish     |     6
 MSB:Host     |    11
 NMMNH:Ento   |     3
 LINGU:Herp   |    65
 CHAS:Ento    |    37
 DMNS:Para    |    15
 APSU:Herp    |    22
 BYU:Herp     |    58
 OWU:Fish     |     3
(45 rows)

```
Contacts:


 @ebraker
 @bryansmclean
 @ehalverson26
 @Nicole-Ridgwell-NMMNHS
 @mkoo
 @campmlc
 @atrox10
 @DerekSikes
 @mvzhuang
 @jtgiermakowski
 @wellerjes
 @keg34
 @jrpletch
 @acdoll
 @jrdemboski
 @happiah-madson
 @genevieve-anderegg
 @byuherpetology
 @msbparasites
 @mlbowser
 @falco-rk
 @ewommack
 @kderieg322079
 @sharpphyl
 @gracz-UNL
 @lin-fred
 @adhornsby
 @jandreslopez
 @droberts49

**Comment by campmlc** [comment]
I agree with the simplified definition (we don't need to explain the assumptions of ontology recapitulates phylogeny here).

Can we add the following as synonymous search terms for phylum Platyhelminthes and phylum Nematoda? - I'm sure other taxonomic groups will need similar requests.
cysticercus/cysticerci
procercoid/procerci
plerocercoid/plurocerci
miracidium/miracidia
sporocyst
redia(e)
cerearia(e)
metacercaria(e)
microfilaria(e)

**Comment by DerekSikes** [comment]
If larva is being moved from age class to life stage, what is replacing it in age class? Maybe I missed this above.

**Comment by dustymc** [comment]
>add the following as synonymous search terms 

Yes, but not in that form. ``cysticercus/cysticerci`` and ``metacercaria(e)`` and etc. should be flattened out into multiple terms - just do whatever it necessary to have one term per row and it'll fit.

**Comment by Jegelewicz** [comment]
>  should be flattened out into multiple terms

Already done in the original request, but someone should review to make sure i didn't screw it up.

**Comment by Jegelewicz** [comment]
> what is replacing it in age class

nothing - we are working to get rid of age class - https://github.com/ArctosDB/arctos/issues/7298

**Comment by campmlc** [comment]
> > should be flattened out into multiple terms
> 
> Already done in the original request, but someone should review to make sure i didn't screw it up.

Done

**Comment by falco-rk** [comment]
Can you add phyllosoma and megalops to the synonyms/search terms? Other than that, I'm happy with the simplified definition.

**Comment by DerekSikes** [comment]
what about juvenile (for spiders), nymph, (for hemimetabolous taxa), protonymph, deuteronymph, tritonymph (for mites); I'm not sure what's happening to the original data - are all these going to be replaced by larva?

**Comment by Jegelewicz** [comment]
> what about juvenile (for spiders), nymph, (for hemimetabolous taxa), protonymph, deuteronymph, tritonymph (for mites); I'm not sure what's happening to the original data - are all these going to be replaced by larva?

I expect juvenile to exist in the code table eventually. As for the others, they aren't included in this issue, but your comment makes me think we could subsume them into this term? Do you agree?

**Comment by Jegelewicz** [comment]
> egg, cocoon, chrysalis, exuvium, larval retreat, pupal exuvium, pupa, puparium, and teneral

The main issue is https://github.com/ArctosDB/arctos/issues/2881 but we are tackling terms one at a time. Your input would be greatly appreciated. Our current plan can be found at https://docs.google.com/spreadsheets/d/1Yv3Lx6DtSdmh0P1cSIG6zW-DnYI6VDJpOkFcgQL6F8o/edit#gid=211205961

**Comment by DerekSikes** [comment]
Can you remind me of why this is happening? What advantage is there in losing precision? Egg & embryo are two things (one can be an empty egg shell without an embryo, the other could be an embryo without an egg case). What advantage do we gain by losing precision? And what about my question regarding functionality? Will lab techs during data entry get a long list of options to pick from that have terms like exuvium or egg on it?

**Comment by mkoo** [comment]
> Can you remind me of why this is happening? What advantage is there in losing precision? Egg & embryo are two things (one can be an empty egg shell without an embryo, the other could be an embryo without an egg case). What advantage do we gain by losing precision? And what about my question regarding functionality? Will lab techs during data entry get a long list of options to pick from that have terms like exuvium or egg on it?

a few reasons aside from just trying to make attributes make sense: 1) align and conform to DarwinCore metadata standards i.e. life_stages; 2) logic of egg and embryo as actually different parts which you point out in your example.  You should be able to unambiguously indicate "an empty egg shell without an embryo" or the reverse or indicate that both exist in the specimen by picking parts and not life stage; 3) shorter more logical pick list for data entry will be the result of this cleanup work. Exuvium and egg should be in parts. Can you help with terms like this to help with the migration (which has been in the works since 2021)? happy to a call on this if easier. just let me know @DerekSikes

**Comment by DerekSikes** [comment]
ah, makes sense to have some of these terms as different parts rather than lifestage - just hope we don't lose any terms that are in use. So, if I choose 'exuvium' as a part I could then choose "pupa" or "larva" as the life stage. Good.

**Comment by mkoo** [comment]
> ah, makes sense to have some of these terms as different parts rather than lifestage - just hope we don't lose any terms that are in use. So, if I choose 'exuvium' as a part I could then choose "pupa" or "larva" as the life stage. Good.

Exactly! Also we wont lose terms since we are also applying synonyms so searching by 'metamorph' should bring up 'juveniles (that's a herp example off the top of my head). 

in the migration, we can note the original term in the remarks too.

**Comment by dustymc** [comment]
term added (but with a filtered bestfor)

**Comment by dustymc** [comment]
added access for collections:

 {ALMNH:Ento,APSU:Fish,APSU:Herp,ASUMZ:Invert,BYU:Herp,CHAS:Ento,CHAS:Herp,CHAS:Inv,CSULB:Herp,DMNS:Inv,DMNS:Para,HWML:Para,KNWR:Ento,LINGU:Herp,MLZ:Fish,MSB:Herp,MSB:Host,MSB:Para,MVZ:Herp,NMMNH:Ento,NMMNH:Paleo,NMU:Para,OGL:Genomic,OWU:Amph,OWU:Fish,OWU:Inv,PSM:Paleo,UAM:Ento,UAM:Herp,UAMObs:Ento,UCM:Fish,UCM:Herp,UCM:Obs,UCSC:Herp,UMNH:Herp,UMZM:Para,UNCG:Para,UNR:Fish,UNR:Herp,USNPC:Para,UTEP:Herp,UTEP:HerpOS,UWBM:Herp,UWYMV:Herp,UWZM:Herp}

**Comment by dustymc** [comment]
done
